### PR TITLE
Multiple datasets to one output

### PR DESCRIPTION
**Pull request by benjwadams**

### CODE REVIEW

**Comment by benjwadams** [comment]
Note: this should be checked against compliance checker plugins prior to merging.

**Comment by benjwadams** [comment]
`Command exited with code -1073741819`

Wat

**Comment by Bobfrat** [comment]
@ocefpaf Do you know what this Appveyor error message means?

**Comment by ocefpaf** [comment]
AppVeyor updated its miniconda for Python 2.7 and broke things :unamused: ignore it for now, I'm working on a workaround.

File: README.md [hunk 1]
```diff
@@ -131,14 +131,17 @@ optional arguments:
   -h, --help            show this help message and exit
   --test TEST, -t TEST, --test= TEST, -t= TEST
                         Select the Checks you want to perform. Defaults to
-                        'acdd' if unspecified
+                        'acdd' if unspecified. Versions of standards can be
+                        specified via `-t <test_standard>:<version>`. If
+                        `<version>` is omitted, or is "latest", the latest
+                        version of the test standard is used.
```
Comment (Bobfrat) [inline README.md]: 👍

File: README.md [hunk 2]
```diff
@@ -148,6 +151,53 @@ optional arguments:
   -d DOWNLOAD_STANDARD_NAMES, --download-standard-names DOWNLOAD_STANDARD_NAMES
                         Specify a version of the cf standard name table to
                         download as packaged version
+(cchecker) [badams@localhost]~/devel/compliance-checker% python cchecker.py --help
```
Comment (Bobfrat) [inline README.md]: This block below seems to be redundant to the section above. Can we combine the 2 and drop the ```[badams@localhost]~```

Comment (daf) [reply README.md]: Common approach would be to use `$ python cchecker.py --help`

File: compliance_checker/runner.py [hunk 3]
```diff
@@ -132,21 +152,37 @@ def html_output(cls, cs, score_groups, output_filename, ds_loc, limit):
...
         '''
         Generates JSON output for the ocmpliance score(s)
         @param cs              Compliance Checker Suite
         @param score_groups    List of results
         @param output_filename The file path to output to
-        @param ds_loc          Location of the source dataset
-        @param limit           The degree of strictness, 1 being the strictest, and going up from there.
+        @param ds_loc          List of source datasets
+        @param limit           The degree of strictness, 1 being the strictest,
+                               and going up from there.
```
Comment (Bobfrat) [inline compliance_checker/runner.py]: Can you add the new param output_type?

File: README.md [hunk 4]
```diff
@@ -131,14 +131,17 @@ optional arguments:
...
+                        `<version>` is omitted, or is "latest", the latest
+                        version of the test standard is used.
   --criteria [{lenient,normal,strict}], -c [{lenient,normal,strict}]
                         Define the criteria for the checks. Either Strict,
                         Normal, or Lenient. Defaults to Normal.
   --verbose, -v         Increase output. May be specified up to three times.
   --skip-checks SKIP_CHECKS, -s SKIP_CHECKS
                         Specifies tests to skip
-  -f {text,html,json}, --format {text,html,json}
+  -f {text,html,json,json_new}, --format {text,html,json,json_new}
```
Comment (Bobfrat) [inline README.md]: So json_new isn't backwards compatible but it supports multiple datasets. The plan would be to replace the current json format with the json_new in a future release (maybe 4.0?)

I tested and it works well.

File: README.md (hunk 2 above)
Comment (Bobfrat) [reply README.md]: @benjwadams I have a task to update the README so I'll merge so I can build off this and I'll fix the couple items left here.

File: README.md (hunk 4 above)
Comment (benjwadams) [reply README.md]: Yeah, ultimately I didn't want to have surprising behavior for applications that might rely on the old JSON output.

File: README.md (hunk 2 above)
Comment (benjwadams) [reply README.md]: Oops, sorry.  Hasty copypasta

File: compliance_checker/runner.py [hunk 5]
```diff
@@ -47,13 +50,25 @@ def run_checker(cls, ds_loc, checker_names, verbose, criteria,
...
         cs = CheckSuite()
-        ds = cs.load_dataset(ds_loc)
+        # using OrderedDict is important here to preserve the order
+        # of multiple datasets which may be passed in
+        score_dict = OrderedDict()
+        if not isinstance(ds_loc, six.string_types):
+            locs = ds_loc
+        # if single dataset, put in list
+        else:
+            locs = [ds_loc]
```
Comment (Bobfrat) [inline compliance_checker/runner.py]: ok so this keeps backwards compatibility by not requiring a list to be passed in

**Comment by benjwadams** [comment]
Ignore the appveyor failure here.  I cherry-picked @ocefpaf's earlier appveyor commit to this branch and it should be functional.

**Pull request merged**